src/Functions/FunctionsStringSimilarity.h

```diff
@@ -132,4 +133,163 @@ class FunctionsStringSimilarity : public IFunction
+
+        for (const auto & key : keys)
+        {
+            if (key.starts_with(string_similarity_model_config_key)) {
+                auto model_name = config.getString(
+                    fmt::format("{}.{}.{}", string_similarity_models_config_key, key, model_name_config_key));
+                auto path_to_positive = config.getString(
+                    fmt::format("{}.{}.{}", string_similarity_models_config_key, key, path_to_positive_config_key));
+                auto path_to_negative = config.getString(
+                    fmt::format("{}.{}.{}", string_similarity_models_config_key, key, path_to_negative_config_key));
+                StringSimilarityModel model {model_name, std::move(path_to_positive), std::move(path_to_negative)};
+                models.emplace(std::move(model_name), std::move(model));
+            }
+        }
+
+        return std::make_shared<FunctionsStringCategorySimilarity>(std::move(models));
+    }
+
+    String getName() const override { return name; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    size_t getNumberOfArguments() const override { return 2; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!isString(arguments[0]))
+            throw Exception(
+                "Illegal type " + arguments[0]->getName() + " of argument of function " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (!isString(arguments[1]))
+            throw Exception(
+                "Illegal type " + arguments[1]->getName() + " of argument of function " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeNumber<typename Impl::ResultType>>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t /*input_rows_count*/) const override
+    {
+        using ResultType = typename Impl::ResultType;
+
+        const ColumnPtr & column_haystack = arguments[0].column;
+        const ColumnPtr & column_needle = arguments[1].column;
+
+        const ColumnConst * col_haystack_const = typeid_cast<const ColumnConst *>(&*column_haystack);
+        const ColumnConst * col_needle_const = typeid_cast<const ColumnConst *>(&*column_needle);
+
+        if (!col_needle_const) {
+            throw Exception("Illegal search model " + arguments[1].column->getName(), ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        const String & model_name = col_needle_const->getValue<String>();
+        const auto& model_it = string_similarity_models.find(model_name);
+        if (model_it == string_similarity_models.end()) {
+            throw Exception("There is no " + model_name + " string similarity model", ErrorCodes::BAD_ARGUMENTS);
+        }
+        const auto& model = model_it->second;
+
+        const String positive_needle = readModel(model.path_to_positive);
```

Comment:
The models are currently loaded again for each block.
Instead, they should be managed by Context, and reloaded similarly to Dictionaries.